src/WebJobs.Extensions.DurableTask.Analyzers/SyntaxNodeUtils.cs

```diff
@@ -46,6 +47,56 @@ public static bool IsInsideOrchestrator(SyntaxNode node)
 
             return false;
         }
+        
+        public static bool IsMarkedDeterministic(SyntaxNode node)
+        {
+            if (TryGetDeterministicAttribute(node, out SyntaxNode deterministicAttribute))
+            {
+                return true;
+            }
+            return false;
+        }
+
+
+        public static bool TryGetDeterministicAttribute(SyntaxNode node, out SyntaxNode deterministicAttribute)
+        {
+            if (TryGetMethodDeclaration(node, out SyntaxNode methodDeclaration))
+            {
+                var IEnumeratorAttributeList = methodDeclaration.ChildNodes().Where(x => x.IsKind(SyntaxKind.AttributeList));
```

Comment:
nit: variable names should start with a lowercase letter. I simple name like `attributeLists` would be fine too.